### PR TITLE
feat: vcs structured logging

### LINKED ISSUE #869: Support Structured Logging
Current logging uses addhoc messages. Add support for structured logging.

### PR DESCRIPTION
**Pull request by Baha-sk**
closes #869

Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: cmd/common/storage.go [hunk 1]
```diff
@@ -171,16 +172,16 @@ func parseURL(u string) (string, string, error) {
...
 
 	return backoff.RetryNotify(
 		task,
 		backoff.WithMaxRetries(backoff.NewConstantBackOff(sleep), numRetries),
 		func(retryErr error, t time.Duration) {
-			logger.Warnf(
-				"failed to connect to storage, will sleep for %s before trying again : %s\n",
-				t, retryErr)
+			logger.Warn(
+				fmt.Sprintf("Failed to connect to storage, will sleep for %s before trying again.",
```
[the remark this answers is not in the document]
Comment (Baha-sk) [reply cmd/common/storage.go]: updated

File: component/event/subscriber.go [hunk 2]
```diff
@@ -60,29 +61,32 @@ func (h *Subscriber) start() {
...
 		case e, ok := <-h.eventChan:
 			if !ok {
-				logger.Infof("event channel closed")
+				logger.Info("event channel closed")
 
 				return
 			}
 
-			logger.Debugf("received new event[%s]: %s", e.ID, string(*e.Data))
+			logger.Debug("received new event", log.WithAdditionalMessage(fmt.Sprintf(
```
[the remark this answers is not in the document]
Comment (Baha-sk) [reply component/event/subscriber.go]: separated now

**Review by fqutishat — APPROVED** [review]

File: pkg/service/oidc4vc/oidc4vc_service.go [hunk 3]
```diff
@@ -134,7 +134,7 @@ func (s *Service) fetchWellKnownConfig(ctx context.Context, url string) (*wellKn
 
 	defer func() {
 		if closeErr := resp.Body.Close(); closeErr != nil {
-			logger.Errorf("Failed to close response body: %s", closeErr)
+			logger.Error("Failed to close response body: %s", log.WithError(closeErr))
```
Comment (bstasyszyn) [inline pkg/service/oidc4vc/oidc4vc_service.go:137]: The %s should be removed from the message.

File: test/bdd/bddtests_test.go [hunk 4]
```diff
@@ -122,11 +123,11 @@ func afterSuiteHook() {
 
 	dockerComposeDown := []string{"docker-compose", "-f", composeFilePath, "down"}
 
-	logger.Infof("Running %s", strings.Join(dockerComposeDown, " "))
+	logger.Info(fmt.Sprintf("Running %s", strings.Join(dockerComposeDown, " ")))
 
 	cmd := exec.Command(dockerComposeDown[0], dockerComposeDown[1:]...) //nolint:gosec
 	if out, err := cmd.CombinedOutput(); err != nil {
-		logger.Fatalf("%s: %s", err.Error(), string(out))
+		logger.Fatal(fmt.Sprintf("%s: %s", err.Error(), string(out)))
```
Comment (bstasyszyn) [inline test/bdd/bddtests_test.go:130]: Should use WithError and another field for "out".

File: test/bdd/pkg/vc-devapi/vc_devapi_steps.go [hunk 5]
```diff
@@ -87,7 +93,8 @@ func (s *Steps) httpGet(url string, withAuth bool) error {
 	s.responseBody = body
 	s.responseStatus = resp.StatusCode
 
-	logger.Infof("body : %v and status %v", string(body), s.responseStatus)
+	logger.Info("httpGet content", log.WithAdditionalMessage(fmt.Sprintf("body: %s", string(body))),
```
Comment (bstasyszyn) [inline test/bdd/pkg/vc-devapi/vc_devapi_steps.go:96]: AdditionalMessage seems to be a bucket for leftovers. I think you should use WithResponseBody(body) here.

**Pull request merged**